README.md

```diff
@@ -0,0 +1,46 @@
+# MWixnet
+This is an implementation of @tromp's [CoinSwap Proposal](https://forum.grin.mw/t/mimblewimble-coinswap-proposal/8322) with some slight modifications.
+
+A set of n CoinSwap servers (node<sub>i</sub> with i=1...n) are agreed upon in advance. They each have a known public key.
+
+### SWAP API
+The first CoinSwap server (n<sub>1</sub>) provides the `swap` API, publicly available for use by GRIN wallets.
+
+**jsonrpc:** `2.0`
+**method:** `swap`
+**params:**
+```
```

Comment:
How do these params relate to the data provisioning below? I don't understand why there is a proof of ownership but no rangeproof. And what is data?

Reply:
`data` refers to the array of encrypted payloads P<sub>i</sub>...P<sub>n</sub>. Maybe `payloads` would be a better name.

Reply:
I had concluded that data was too short (at a mere 34 bytes) to contain a rangeproof:-(

Reply:
True, I should've included a rangeproof in the payload. I will change to `payloads` and replace this example with a real one for 3 nodes. We'll see 2 of these small payloads followed by a larger one with the rangeproof included.

I wasn't sure what message should be signed for the ComSig, so I just added a msg field. I guess the Onion field could be signed instead. Thoughts?

Reply:
I can't think of a useful message, so we could keep it empty for now...